src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,98 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private static final double MULTIPLIER = 1.5;
+    private int size;
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        growIfSizeFull();
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        indexVerifyForAdd(index);
+        growIfSizeFull();
+        System.arraycopy(array, index, array, index + 1, size - index);
+        array[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexVerify(index);
+        return array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        indexVerify(index);
+        array[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexVerify(index);
+        T removedObject = array[index];
+        System.arraycopy(array, index + 1, array, index,size - index - 1);
+        size--;
+        return removedObject;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < array.length; i++) {
```

Comment:
In the remove(T element) method, it's better to use 'size' instead of 'array.length' in the for loop condition, to avoid unnecessary iterations and possible mistakes due to leftover values after removing elements